### PR TITLE
Controller improvements

### PR DESCRIPTION
**Pull request by maltesander**
Moved 2 methods that are used in every operator into operator-rs.

### CODE REVIEW

File: src/role_utils.rs [hunk 1]
```diff
@@ -118,3 +121,112 @@ pub async fn find_nodes_that_fit_selectors(
     }
     Ok(found_nodes)
 }
+
+/// For each role, return a tuple consisting of eligible nodes for a given selector.
+/// Required to delete excess pods that do not match any node or selector description.
+pub fn get_full_pod_node_map(
```
[the remark this answers is not in the document]
Comment (lfrancke) [reply src/role_utils.rs:127]: No better suggestion from me.... other than something like `eligible_nodes_for_role` or something along those lines.

But (without looking at the code) is the comment correct? Does it do so for each "role" or for each "role group"?
And the description doesn't say what else is in that tuple. Eligible nodes and....

[the remark this answers is not in the document]
Comment (soenkeliebau) [reply src/role_utils.rs:127]: The tuple contains eligible nodes and a selector - that is two things. 
 I had to read it twice as well though, so maybe it wasn't just me that found it hard to grok :)

[the remark this answers is not in the document]
Comment (maltesander) [reply src/role_utils.rs:127]: I have no real idea. Im fine with both.

File: src/role_utils.rs [hunk 2]
```diff
@@ -118,3 +121,112 @@ pub async fn find_nodes_that_fit_selectors(
...
+    );
+    labels
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_get_full_pod_node_map() {
```
[the remark this answers is not in the document]
Comment (maltesander) [reply src/role_utils.rs:169]: Yeah i like that, will adapt it.

File: src/role_utils.rs [hunk 3]
```diff
@@ -118,3 +121,112 @@ pub async fn find_nodes_that_fit_selectors(
...
+            }
+
+            if let Some(role_label) = labels.get(labels::APP_COMPONENT_LABEL).unwrap() {
+                assert_eq!(role_label, role);
+            }
+        }
+    }
+
+    #[test]
+    fn test_get_role_and_group_labels() {
```
[the remark this answers is not in the document]
Comment (maltesander) [reply src/role_utils.rs:212]: Not sure, basically we fill a hashmap with 2 entries. We really need to make an rstest out of it?

[the remark this answers is not in the document]
Comment (soenkeliebau) [reply src/role_utils.rs:287]: You are right, yes.. I wasn't paying attention, I thought there was at least an option or something that we could test for..

File: src/role_utils.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (soenkeliebau) [reply src/role_utils.rs:127]: How about something like `get_target_node_selectors`?

[the remark this answers is not in the document]
Comment (maltesander) [reply src/role_utils.rs:127]: Ok i thought about it a little:
@soenkeliebau yeah i think pod does not belong there at all, that connection is made later (check delete_excess_pods)

For `get_target_node_selectors` i think it suggests to get the (possible?) selectors for a target node (which we dont do here).
What this method does is collecting eligible nodes for a role and group (via labels).
 
So the more i think of it i like @lfrancke idea with extension (and checks for role and group):
`list_eligible_nodes_for_role_and_group`

I can however live with just `get_full_node_map` which is kinda true, just not very specific.

[the remark this answers is not in the document]
Comment (soenkeliebau) [reply src/role_utils.rs:127]: Either one is fine for me

**Review by soenkeliebau — APPROVED** [review]
LGTM

**Pull request merged**